### PR TITLE
migrate scheduling logic to schedule_one.go as more reasonable

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup
/sig scheduling

#### What this PR does / why we need it:
With this, scheduler.go would be the file the includes the instantiation logic, while schedule_one.go includes the scheduling logic.: https://github.com/kubernetes/kubernetes/pull/108790#issuecomment-1073282141
#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/kubernetes/issues/108788

#### Special notes for your reviewer:
Just code migration, no other changes  except one place I added comment already.

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/factory_test.go [hunk 1]
```diff
@@ -15,308 +15,3 @@ limitations under the License.
...
-	extenderv1 "k8s.io/kube-scheduler/extender/v1"
-	"k8s.io/kubernetes/pkg/scheduler/framework"
-	internalcache "k8s.io/kubernetes/pkg/scheduler/internal/cache"
-	internalqueue "k8s.io/kubernetes/pkg/scheduler/internal/queue"
-	testingclock "k8s.io/utils/clock/testing"
-)
-
-const (
-	podInitialBackoffDurationSeconds = 1
-	podMaxBackoffDurationSeconds     = 10
```
Comment (kerthcet) [inline pkg/scheduler/factory_test.go:43]: Removed `podInitialBackoffDurationSeconds` and `podMaxBackoffDurationSeconds` as we don't use them in any place.

File: pkg/scheduler/scheduler_one_test.go [hunk 2]
```diff
@@ -0,0 +1,2864 @@
+/*
```
Comment (denkensk) [inline pkg/scheduler/scheduler_one_test.go:1]: pkg/scheduler/scheduler_one_test.go  --> pkg/scheduler/schedule_one_test.go

Comment (kerthcet) [reply pkg/scheduler/scheduler_one_test.go]: done :)

File: pkg/scheduler/schedule_one.go [hunk 3]
```diff
@@ -0,0 +1,866 @@
...
+	return ScheduleResult{
+		SuggestedHost:  host,
+		EvaluatedNodes: len(feasibleNodes) + len(diagnosis.NodeToStatusMap),
+		FeasibleNodes:  len(feasibleNodes),
+	}, err
+}
+
+// snapshot snapshots scheduler cache and node infos for all fit and priority
+// functions.
+func (sched *Scheduler) snapshot() error {
```
Comment (ahg-g) [inline pkg/scheduler/schedule_one.go:364]: we can remove this function and just invoke `sched.Cache.UpdateSnapshot(sched.nodeInfoSnapshot)` directly at line 315 above.

Comment (kerthcet) [reply pkg/scheduler/schedule_one.go:364]: done :)

**Pull request merged**